includes/cache/wikia/CeleryPurge.class.php

```diff
@@ -71,6 +71,30 @@ static function onRestInPeace() {
 	}
 
 	/**
+	 * SUS-81: allow CDN purging by surrogate key
+	 *
+	 * Use OutputPage::tagWithSurrogateKeys() helper to emit proper headers
+	 *
+	 * @param string $key surrogate key to purge
+	 * @param string $service Fastly's service name (defaults to mediawiki)
+	 */
+	static function purgeBySurrogateKey( $key, $service = 'mediawiki' ) {
+		$caller = self::getPurgeCaller();
+		wfDebug( "Purging backtrace: " . wfGetAllCallers( false ) . "\n" );
+
+		WikiaLogger::instance()->info( 'varnish.purge', [
+			'key' => $key,
+			'service' => $service
+		] );
+
+		( new AsyncCeleryTask() )
+			->taskType( 'celery_workers.purger.purge' )
```

Comment:
Oh. As I look at this more I'm not sure what it's doing. What is the difference between this and the [purge](https://github.com/Wikia/app/pull/9663/files#diff-a3b45c6089cdc13c1ebda814abf04577R221) above?